src/main/java/com/ciandt/techgallery/service/impl/EmailServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+        }
+      }
+    }
+    TechGalleryUser user = TechGalleryUserDAOImpl.getInstance().findById(Long.parseLong(userId));
+    Technology technology = TechnologyDAOImpl.getInstance().findById(technologyId);
+
+    // TODO extract method to build template (mustache).
+    Map<String, String> variableValue = new HashMap<String, String>();
+    variableValue.put("${receiverName}", user.getName());
+    variableValue.put("${user}", user.getName());
+    variableValue.put("${technology}", technology.getName());
+    EmailConfig email = new EmailConfig(subject, "emailtemplates" + File.separator + template,
+        variableValue, null, reason, user.getEmail());
+    sendEmail(email);
+  }
+
+  private void sendEmail(EmailConfig email) {
+    try {
+      Message msg = prepareMessage(email);
+      Transport.send(msg);
+      registerEmailNotification(email, true);
+
+    } catch (Throwable err) {
+      Long notificationId = registerEmailNotification(email, false);
+      log.log(Level.SEVERE, "Error when attempting to send email " + notificationId, err);
+    }
+  }
+
+  private Message prepareMessage(EmailConfig email)
+      throws UnsupportedEncodingException, MessagingException {
+
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
+    Message msg = new MimeMessage(session);
+    msg.setFrom(getFrom());
+    for (String to : email.getTo()) {
+      msg.addRecipient(Message.RecipientType.TO, new InternetAddress(to));
+    }
+    msg.setSubject(email.getSubject());
+    email.processTemplate();
+    msg.setText(email.getBody());
+    return msg;
+  }
+
+  private long registerEmailNotification(EmailConfig email, boolean success) {
+    EmailNotification emailNotification = new EmailNotification();
+    emailNotification.setRecipients(Arrays.asList(email.getTo()));
+    emailNotification.setRule(email.getRule());
+    emailNotification.setReason(email.getReason());
+    emailNotification.setTimestampSend(System.currentTimeMillis());
+    emailNotification.setEmailStatus(success ? "SUCCESS" : "FAILURE");
+
+    Key<EmailNotification> key = emailNotificationDao.add(emailNotification);
+    return key.getId();
+  }
+
+  private InternetAddress getFrom() throws UnsupportedEncodingException {
+    if (from == null) {
+      String appId = ApiProxy.getCurrentEnvironment().getAppId();
+      int tilde = appId.indexOf('~');
+      if (tilde >= 0) { // TODO make this into an environment property
+        appId = appId.substring(tilde + 1);
+      }
+      String addr = "google-project@ciandt.com";
+      log.info("app email from address set to: " + addr);
+      from = new InternetAddress(addr, "no-reply@google.com");
+    }
+    return from;
+  }
+
+  /**
+   * Push one email to email queue for each follower user in each technology that have
+   * followers.
+   */
+  public void sendEmailtoFollowers() {
+    CronJob cronJob = new CronJob();
```

Comment:
@brunoliberal EmailService should only send emails. Create a service class for this cron job.